src/main/webapp/src/app/nav-bar/nav-bar.component.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { ComponentFixture, TestBed, inject } from '@angular/core/testing';
+import { NavBarComponent } from './nav-bar.component';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { By } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MapComponent } from '../map/map.component';
+
+describe('NavBarComponent', () => {
+ 
+  let component: NavBarComponent;
+  let fixture: ComponentFixture<NavBarComponent>;
+  
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ RouterTestingModule ],
+      declarations: [ NavBarComponent ],
+      providers: [
+        NavBarComponent
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        CommonModule,
+        RouterTestingModule.withRoutes([
+         { path: '', component: MapComponent }
+        ])
+      ],
+    }),
+    
+    fixture = TestBed.createComponent(NavBarComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create a NavBar', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should go to home url',
+    inject([Router, Location], (router: Router, location: Location)=> {
```

Comment:
you don't need to inject the router in here as you're not using it in this test.
you could also argue you don't necessarily need to inject the location since you're not making a custom location or anything, you could simply do something like location = TestBed.get(Location); in the before each function and then use that variable in this test.